### PR TITLE
Summary stats

### PR DESCRIPTION
**Pull request by RianneJ**
Add unit tests, make changes to summary.R script facilitate unit testing

### CODE REVIEW

File: web-app/HeimScripts/heatmap/summary.R [hunk 1]
```diff
@@ -29,14 +29,43 @@
...
-main <- function(phase)
+main <- function(phase = NULL)
 {
+  check_input(loaded_variables, phase)
   data_measurements <- extract_measurements(loaded_variables)
-  produce_summary_stats(data_measurements, phase)
+  summary_stats_json <- produce_summary_stats(data_measurements, phase)
+  write_summary_stats(summary_stats_json)
   produce_boxplot(data_measurements, phase)
   return(list(summary_stats = "Finished")) #right now a non-empty list is expected as a return.
```
Comment (PiotrZakrzewski) [inline web-app/HeimScripts/heatmap/summary.R]: Current agreement with the Frontend is that we pass "messages" list. A vector of Strings. These messages will be displayed on the frontend and they also serve as nice filler for the list we have to return from main function anyway.

File: web-app/HeimScripts/heatmap/summary.R [hunk 2]
```diff
@@ -29,14 +29,43 @@
...
+  names_in_correct_format <- grepl(expected_format_names, dataset_names)
+  if(any(!names_in_correct_format))
+  {
+    stop(paste("One or more labels of the datasets do not have the expected format.", 
+               "\nExpected format: an unique numerical identifier for the node appended behind the letter \"n\",\nfollowed by an underscore and an unique numerical identifier for the subset appended behind an \"s\",",
+               "\ne.g. n0_s1, n0_s2, n1_s1, n1_s2. "))
+  }
+  if(is.null(phase_info))
+  {
+    stop("supply phase parameter to function \'main()\'")
```
Comment (PiotrZakrzewski) [inline web-app/HeimScripts/heatmap/summary.R]: If we use stop - Frontend will just return unspecific error, without any meaningful message for the user. 
Would it not be better to : 
- set messages variable with appropriate message for the user
- do not call stop, let the script return (just without the boxplot of course)

File: web-app/HeimScripts/heatmap/summary.R [hunk 3]
```diff
@@ -15,8 +15,8 @@
 #     are not overwritten if the script is run for the 'preprocess data' tab)
```
Comment (PiotrZakrzewski) [inline web-app/HeimScripts/heatmap/summary.R]: You do not make use of "preprocessed" variable right? Is it intended? How will you generate summary stats for the preprocessed data?

File: web-app/HeimScripts/heatmap/summary.R [hunk 4]
```diff
@@ -147,6 +198,9 @@ produce_boxplot <- function(measurement_tables, phase)
     measurement_vectors[[i]] <- unlist(measurement_vectors[[i]])
```
Comment (PiotrZakrzewski) [inline web-app/HeimScripts/heatmap/summary.R:230]:
```
  nodes <- gsub("_.*","",names(measurement_tables))
  subsets <- gsub(".*_","", names(measurement_tables))
```

Are nodes already included in the input?

File: web-app/HeimScripts/heatmap/summary.R (hunk 3 above)
Comment (RianneJ) [inline web-app/HeimScripts/heatmap/summary.R]: That was my next step. Will implement it later

File: web-app/HeimScripts/heatmap/summary.R (hunk 4 above)
Comment (RianneJ) [inline web-app/HeimScripts/heatmap/summary.R:230]: Yes, in loaded variables the label of each dataframe is of format n<node_identifier>_s<subset_identifier>, where the identifiers are numerical
For the variable preprocessed I do not know

**Pull request merged**